stimuli_ci_generation/src/server/test_ci.py

```diff
@@ -0,0 +1,70 @@
+import os
+import main
+import gcp_config
+from google.cloud import storage
+from server.gcp_cloud_storage import upload_file
+
+# NUM_STIMULI_FILES = (number of images per selection) * (number of selections) + number of rdata file
+NUM_STIMULI_FILES = 2 * 200 + 1
+
+
+def test_stimuli_ci(test_asset_path="./test_asset"):
+    storage_client = storage.Client()
+
+    # delete cached test files in bucket
+    for identifier in os.listdir(test_asset_path):
+        path = test_asset_path + identifier
+        if os.path.isdir(path):
+            for blob in storage_client.list_blobs(
+                gcp_config.MASKED_IMG_BUCKET, prefix=identifier
+            ):
+                blob.delete()
+            for blob in storage_client.list_blobs(
+                gcp_config.STIMULI_IMG_BUCKET, prefix=identifier
+            ):
+                blob.delete()
+            for blob in storage_client.list_blobs(
+                gcp_config.USER_SELECTION_BUCKET, prefix=identifier
+            ):
+                blob.delete()
+            for blob in storage_client.list_blobs(
+                gcp_config.CI_IMG_BUCKET, prefix=identifier
+            ):
+                blob.delete()
+
+    start_time = time.time()
+
+    for identifier in os.listdir(test_asset_path):
+        path = test_asset_path + identifier
+        if os.path.isdir(path):
+            test_start_time = time.time()
+
+            upload_file(
+                gcp_config.MASKED_IMG_BUCKET,
+                test_asset_path + identifier + "masked.jpg",
+                identifier + "neutral.jpg",
+            )
+            stimuli_results = list(
+                storage_client.list_blobs(
+                    gcp_config.STIMULI_IMG_BUCKET, prefix=identifier
+                )
+            )
+            while len(stimuli_results) < NUM_STIMULI_FILES:
+                sleep(100)
```

Comment:
you can use `onSnapshot` to listen to real time changes to firestore so we don't need to sleep here

source: https://firebase.google.com/docs/firestore/query-data/listen

Reply:
does this work with not yet created objects?

Reply:
What do you mean by not yet created? To clarify, once the stimuli have been generated, our cloud run instance will modify the corresponding user doc and we can listen to that as a completion signal. Pls let me know if I'm (mis)understanding what you are doing here

Reply:
Ah i see why you mean, but this will not always work unless there's a user document which is not guaranteed especially with arbitrary tests

Reply:
should we make a test user hmm? btw I took another look and wondered if the while loop might be an infinite loop? Does 
```
         stimuli_results = list(
                storage_client.list_blobs(
                    gcp_config.STIMULI_IMG_BUCKET, prefix=identifier
                )
            )
```
get real-time updated if there more items in the bucket?

Reply:
good catch, its fixed now